library/alloc/src/sync.rs

```diff
@@ -2364,7 +2364,7 @@ unsafe fn data_offset<T: ?Sized>(ptr: *const T) -> isize {
     // Because it is `?Sized`, it will always be the last field in memory.
     // Note: This is a detail of the current implementation of the compiler,
     // and is not a guaranteed language detail. Do not rely on it outside of std.
-    unsafe { data_offset_align(align_of_val(&*ptr)) }
+    unsafe { data_offset_align(align_of_val_raw(&*ptr)) }
```

Comment:
This still materializes a reference. Shouldn't it be `align_of_val_raw(ptr)`?

Reply:
~~Yes, you saw nothing~~

That's what I get for writing this quickly 🙃